controllers/controllers/resource/reconciler.go

```diff
@@ -96,6 +102,41 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 			return err
 		}
 		resources = append(resources, r...)
+	case anywherev1.CloudStackDatacenterKind:
+		csdc := &anywherev1.CloudStackDatacenterConfig{}
+		cpCsmc := &anywherev1.CloudStackMachineConfig{}
+		etcdCsmc := &anywherev1.CloudStackMachineConfig{}
+		workerCsmc := &anywherev1.CloudStackMachineConfig{}
+		workerCsmcs := make(map[string]anywherev1.CloudStackMachineConfig, len(cs.Spec.WorkerNodeGroupConfigurations))
+		err := cor.FetchObject(ctx, types.NamespacedName{Namespace: objectKey.Namespace, Name: cs.Spec.DatacenterRef.Name}, csdc)
+		if err != nil {
+			return err
+		}
+		err = cor.FetchObject(ctx, types.NamespacedName{Namespace: objectKey.Namespace, Name: cs.Spec.ControlPlaneConfiguration.MachineGroupRef.Name}, cpCsmc)
+		if err != nil {
+			return err
+		}
+		if len(cs.Spec.WorkerNodeGroupConfigurations) != 1 {
```

Comment:
so no multiple worker node groups support?

Reply:
I think this was talked about in the past, but I don't remember if the plan was to release without this support. We should keep cloudstack behind a feature flag if we don't support multiple worker node groups imo